src/libs/tools/include/errors/errorTypes.hpp

```diff
@@ -0,0 +1,125 @@
+
+#ifndef ELEKTRA_ERRORTYPES_HPP
+#define ELEKTRA_ERRORTYPES_HPP
+
+#include "error.hpp"
+
+namespace kdb
+{
+namespace tools
+{
+namespace errors
+{
+
+/* Not an error by itself, but a container for multiple warnings */
+class PureWarningError : public Error
+{
+public:
+	PureWarningError ()
+	: Error { "No error, only warnings.", "", "", "", "", 0} {}
+
+	std::string code() const override;
+	std::string description() const override;
+
+private:
+	bool compare(const BaseNotification& other) const final;
+};
+
+
+/* Concrete classes for the different Errors, based on the constants defined in /src/include/kdberrors.h */
+class ResourceError : public Error
```

Comment:
Actually this could even be in src/bindings/cpp/include/kdbexcept.hpp this would need some exception mapping in src/bindings/cpp/include/kdb.hpp, though. (extra work beyond original scope of project)